jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/AStarInconsistentHeuristic.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (C) Copyright 2015-2018, by Brooks Bockman and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.util.*;
+
+import org.jgrapht.*;
+/**
+ * Interface for an admissible but inconsistent heuristic used in A* search.
+ * This interface is based primarily on {@link AStarAdmissibleHeuristic}, but
+ * includes additional methods to update inconsistent heuristics, helping to
+ * optimize A* search with inconsistent heuristics.  This interface is meant
+ * to implement a BPMX style update step.  This update essentially checks the
+ * heuristic values of a vertices neighbors to determine consistency, and if
+ * it is inconsistent the heuristic estimate is increased to the lowest value
+ * that would provide consistency.  Note, this BPMX approach does not fully 
+ * convert a inconsistent heuristic into a consistent one.
+ * 
+ * For a basic implementation of this interface refer to the test repository
+ * class ALTInconsistentHeuristic.
+ *
+ * @param <V> vertex type
+ * @author Brooks Bockman 
+ */
+public interface AStarInconsistentHeuristic<V>
```

Comment:
Without documentation, the method description is too shallow. Most users would not be able to implement this interface as it is quite unclear what exactly is expected and what it is used for.